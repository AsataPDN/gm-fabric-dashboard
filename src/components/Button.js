import { PropTypes } from "prop-types";
import React from "react";

Button.propTypes = {
  active: PropTypes.bool,
  clickAction: PropTypes.any.isRequired,
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  iconSize: PropTypes.oneOf(["normal", "xs", "sm", "lg", "xl"]),
  label: PropTypes.string.isRequired,
  orientation: PropTypes.oneOf(["vertical", "horizontal"]),
  outline: PropTypes.oneOf([
    "raised",
    "outline",
    "outline-shadow",
    "shadow",
    "none"
  ]),
  prefix: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  size: PropTypes.oneOf(["normal", "xs", "sm", "lg", "xl"]),
  suffix: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  tabIndex: PropTypes.number,
  type: PropTypes.oneOf(["danger", "info", "primary", "secondary", "warning"])
};

function Button({
  active,
  clickAction,
  disabled,
  icon,
  iconSize,
  label,
  orientation,
  prefix,
  size,
  suffix,
  outline,
  tabIndex,
  type
}) {
  return (
    <button
      className={
        "btn" +
        (active ? ` active` : "") +
        (type ? ` btn-type-${type}` : "") +
        (iconSize ? ` btn-icon-size-${iconSize}` : "") +
        (size ? ` btn-size-${size}` : "") +
        (outline ? ` btn-outline-${outline}` : "") +
        (orientation ? ` btn-orientation-${orientation}` : "")
      }
      disabled={disabled}
      onClick={clickAction}
      tabIndex={tabIndex}
      title={label}
    >
      {icon && <span className="icon" data-uk-icon={`icon: ${icon};`} />}
      <span className="label">
        {prefix
          ? <span className="label-prefix">
              {prefix}
            </span>
          : ""}
        {label}
        {suffix
          ? <span className="label-suffix">
              {suffix}
            </span>
          : ""}
      </span>
    </button>
  );
}

export default Button;
