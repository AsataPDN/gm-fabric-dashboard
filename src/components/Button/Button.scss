// components/buttons.scss
@import "../../style/variables.scss";
@import "../../style/mixins.scss";

// Local Variables
// Variables for Buttons are contained in button-mixins.scss

// Mixins
@mixin function-button-icon-size(
  $button-icon-size: $button-icon-size-default) {
}

@mixin function-button-orientation(
  $button-content-orientation: $button-content-orientation-default,
  $button-content-spacing-base: $button-content-spacing-base-default
) {
  @if $button-content-orientation == horizontal {
    flex-direction: row;
    > :not(:first-child) {
      margin-left: $button-content-spacing-base;
      flex-basis: auto;
    }
  } @else if $button-content-orientation == vertical {
    padding-top: $button-content-spacing-base;
    flex-direction: column;
    > :not(:first-child) {
      margin-top: $button-content-spacing-base / 2;
    }
  }
}

@mixin button-style-set($base-color: null) {
  &:not([class*="btn-outline"]),
  &.btn-outline-raised-outline {
    @if $base-color {
      @include function-button-style(
        $button-outline-style: raised-outline,
        $button-background-color-base: $base-color
      );
    } @else {
      @include function-button-style(
        $button-outline-style: raised-outline
      );
    }
  }
  &.btn-outline-shadow {
    @if $base-color {
      @include function-button-style(
      $button-outline-style: shadow,
      $button-background-color-base: $base-color
    );
    } @else {
      @include function-button-style(
      $button-outline-style: shadow
    );
    }
  }
  &.btn-outline-raised {
    @if $base-color {
      @include function-button-style(
      $button-outline-style: raised,
      $button-background-color-base: $base-color
    );
    } @else {
      @include function-button-style(
      $button-outline-style: raised
    );
    }
  }
  &.btn-outline-outline {
    @if $base-color {
      @include function-button-style(
      $button-outline-style: outline,
      $button-background-color-base: $base-color
    );
    } @else {
      @include function-button-style(
      $button-outline-style: outline
    );
    }
  }
  &.btn-outline-none {
    @if $base-color {
      @include function-button-style(
      $button-outline-style: none,
      $button-background-color-base: $base-color
    );
    } @else {
      @include function-button-style(
      $button-outline-style: none
    );
    }
  }
}

// Styles
.btn {
  @include function-button-base();

  // Styles
  &:not([class*="btn-type"]),
  &.btn-default {
    @include button-style-set();
  }
  &.btn-type-danger {
    @include button-style-set($color-danger);
  }
  &.btn-type-info {
    @include button-style-set($color-info);
  }
  &.btn-type-warning {
    @include button-style-set($color-warning);
  }
  &.btn-type-primary {
    @include button-style-set($color-brand-primary);
  }
  &.btn-type-polling {
    @include button-style-set($color-content-background);
  }

  // Sizes
  &:not([class*="btn-size"]),
  &.btn-size-normal {
    @include function-button-size();
  }
  &.btn-size-xs {
    @include function-button-size(xs);
  }
  &.btn-size-sm {
    @include function-button-size(sm);
  }
  &.btn-size-lg {
    @include function-button-size(lg);
  }
  &.btn-size-xl {
    @include function-button-size(xl);
  }

  // Icons
  &:not([class*="btn-icon-size"]),
  &.btn-icon-size-normal {
    @include function-button-icon-size ();
  }
  &.btn-icon-size-xs {
    @include function-button-icon-size ($button-icon-size-default * .5);
  }
  &.btn-icon-size-sm {
    @include function-button-icon-size ($button-icon-size-default * .75);
  }
  &.btn-icon-size-lg {
    @include function-button-icon-size ($button-icon-size-default * 1.25);
  }
  &.btn-icon-size-xl {
    @include function-button-icon-size ($button-icon-size-default * 1.5);
  }

  // Orientations
  &:not(.btn-orientation-vertical),
  &.btn-orientation-horizontal {
    @include function-button-orientation(
      $button-content-orientation: horizontal
    );
  }
  &.btn-orientation-vertical {
    @include function-button-orientation(
      $button-content-orientation: vertical
    );
  }
}
