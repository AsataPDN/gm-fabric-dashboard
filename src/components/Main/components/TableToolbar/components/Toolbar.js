import styled from "styled-components";

import {
  COLOR_CONTENT_BACKGROUND,
  CONTENT_MAX_WIDTH
} from "style/styleVariables";
import { contrastColor, spacingScale } from "style/styleFunctions";

const TOOLBAR_BACKGROUND_COLOR = contrastColor(COLOR_CONTENT_BACKGROUND, 0.04)
  .hsl()
  .string();

const Toolbar = styled.div`
  display: flex;
  height: ${spacingScale(4.5)};
  align-items: center;
  flex-direction: row;
  background-color: ${TOOLBAR_BACKGROUND_COLOR};
  padding: ${spacingScale(0.25)} ${spacingScale(2)};
  max-width: ${CONTENT_MAX_WIDTH};
  margin-left: auto;
  margin-right: auto;

  @media all and (min-width: ${CONTENT_MAX_WIDTH}) {
    &:after {
      content: "";
      pointer-events: none;
      height: inherit;
      position: absolute;
      left: 0;
      right: 0;
      top: 0;
      background-color: inherit;
      z-index: 1;
    }

    > * {
      position: relative;
      z-index: 2;
    }
  }
`;

export default Toolbar;
