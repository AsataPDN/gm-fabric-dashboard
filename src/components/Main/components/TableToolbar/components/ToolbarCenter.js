import styled from "styled-components";

const ToolbarCenter = styled.div`
  display: flex;
  flex: 1 1 33%;
  flex-direction: row;
  justify-content: center;
`;

export default ToolbarCenter;
