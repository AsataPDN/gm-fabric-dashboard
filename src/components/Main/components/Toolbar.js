import styled from "styled-components";

import { COLOR_CONTENT_BACKGROUND } from "style/styleVariables";
import { contrastColor, spacingScale } from "style/styleFunctions";

const TOOLBAR_BACKGROUND_COLOR = contrastColor(COLOR_CONTENT_BACKGROUND, 0.04)
  .hsl()
  .string();

const Toolbar = styled.div`
  display: flex;
  height: ${spacingScale(4.5)};
  align-items: center;
  flex-direction: row;
  background-color: ${TOOLBAR_BACKGROUND_COLOR};
  padding: ${spacingScale(0.25)} ${spacingScale(2)};
`;

export default Toolbar;
