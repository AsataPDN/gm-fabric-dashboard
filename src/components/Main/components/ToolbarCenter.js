import styled from "styled-components";

const ToolbarCenter = styled.div`
  flex: 1 1 33%;
  display: flex;
  justify-content: center;
`;

export default ToolbarCenter;
