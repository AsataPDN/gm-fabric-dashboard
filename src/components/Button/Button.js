import { PropTypes } from "prop-types";
import React from "react";
import "./Button.scss";

import Icon from "components/Icon";
import Glyph from "components/Glyphs/";

Button.propTypes = {
  active: PropTypes.bool, // If the button should be style as active or not
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  clickAction: PropTypes.any.isRequired, // click handler
  disabled: PropTypes.bool, // disables the button
  glyph: PropTypes.string, // Glyph to display in the button
  glyphColor: PropTypes.string, // Color for the glyph
  glyphRatio: PropTypes.oneOfType([PropTypes.string, PropTypes.number]), // Relative size for the glyph
  label: PropTypes.string.isRequired, // label for the button
  orientation: PropTypes.oneOf(["vertical", "horizontal"]), // Vertical: Icon top, label bottom; Horizontal: Icon left, label right;
  outline: PropTypes.oneOf([
    "raised", // Add highlight effect to top edge and shadow effect to bottom edge
    "outline", // Add outline effect
    "outline-shadow", // Add outline effect, and add shadow effect to bottom edge
    "shadow", // Add shadow effect to bottom edge
    "none" // No effects
  ]),
  prefix: PropTypes.oneOfType([PropTypes.string, PropTypes.number]), // Add prefix text to button label
  size: PropTypes.oneOf(["normal", "xs", "sm", "lg", "xl"]), // Relative size of the button
  style: PropTypes.object, // style prop
  suffix: PropTypes.oneOfType([PropTypes.string, PropTypes.number]), // Add suffix text to button label
  tabIndex: PropTypes.number,
  type: PropTypes.oneOf([
    "danger",
    "info",
    "primary",
    "secondary",
    "warning",
    "polling"
  ])
};

/**
 * General purpose button
 * @param {Object} props - see propTypes
 * @returns JSX.Element
 */
function Button({
  active,
  children,
  clickAction,
  disabled,
  glyph,
  glyphRatio,
  glyphColor,
  label,
  orientation,
  prefix,
  size,
  style,
  suffix,
  outline,
  tabIndex,
  type
}) {
  return (
    <button
      className={
        "btn" +
        (active ? ` active` : "") +
        (type ? ` btn-type-${type}` : "") +
        (size ? ` btn-size-${size}` : "") +
        (outline ? ` btn-outline-${outline}` : "") +
        (orientation ? ` btn-orientation-${orientation}` : "")
      }
      disabled={disabled}
      onClick={clickAction}
      tabIndex={tabIndex}
      title={label}
      style={style}
    >
      {glyph && (
        <Icon>
          <Glyph glyphColor={glyphColor} name={glyph} ratio={glyphRatio} />
        </Icon>
      )}
      {children}
      <span className="label">
        {prefix ? <span className="label-prefix">{prefix}</span> : ""}
        {label}
        {suffix ? <span className="label-suffix">{suffix}</span> : ""}
      </span>
    </button>
  );
}

export default Button;
