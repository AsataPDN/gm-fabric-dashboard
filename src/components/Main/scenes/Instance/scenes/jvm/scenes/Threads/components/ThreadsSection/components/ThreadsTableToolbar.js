import { Actions } from "jumpstate";
import { PropTypes } from "prop-types";
import React, { Component } from "react";

// Importing external deps from src as WebPack Modules directory
import ButtonGroup from "components/ButtonGroup";
import Button from "components/Button";
import Toolbar from "components/Main/components/TableToolbar/components/Toolbar";
import ToolbarCenter from "components/Main/components/TableToolbar/components/ToolbarCenter";

/**
 * Toolbar containing buttons to control what is rendered in the threads table
 * @class ThreadsTableToolbar
 * @extends {Component}
 */
class ThreadsTableToolbar extends Component {
  static propTypes = {
    threadCounts: PropTypes.object,
    threadsFilter: PropTypes.string
  };
  render() {
    const { threadCounts, threadsFilter } = this.props;
    //TODO: figure out why there's an empty className on the parent DIV
    return (
      <Toolbar>
        <ToolbarCenter>
          <ButtonGroup>
            <Button
              active={threadsFilter === "all"}
              clickAction={() => Actions.setThreadsFilter("all")}
              disabled={!threadCounts.all}
              tabIndex={0}
              label="All Threads"
              suffix={threadCounts.all}
            />

            <Button
              active={threadsFilter === "active"}
              clickAction={() => Actions.setThreadsFilter("active")}
              disabled={!threadCounts.active}
              tabIndex={0}
              label="Active"
              suffix={threadCounts.active}
            />
            <Button
              active={threadsFilter === "idle"}
              clickAction={() => Actions.setThreadsFilter("idle")}
              disabled={!threadCounts.idle}
              tabIndex={0}
              label="Idle"
              suffix={threadCounts.idle}
            />
            <Button
              active={threadsFilter === "stopped"}
              clickAction={() => Actions.setThreadsFilter("stopped")}
              disabled={!threadCounts.stopped}
              tabIndex={0}
              label="Stopped"
              suffix={threadCounts.stopped}
            />
          </ButtonGroup>
        </ToolbarCenter>
      </Toolbar>
    );
  }
}

export default ThreadsTableToolbar;
