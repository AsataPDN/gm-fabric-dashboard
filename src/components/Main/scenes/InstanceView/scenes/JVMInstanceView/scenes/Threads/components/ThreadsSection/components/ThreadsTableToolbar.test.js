import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import ThreadsTableToolbar from "./ThreadsTableToolbar";
import ButtonGroup from "components/ButtonGroup";
import Button from "components/Button";
import Toolbar from "components/Main/components/TableToolbar/components/Toolbar";
import ToolbarCenter from "components/Main/components/TableToolbar/components/ToolbarCenter";

// import Actions
import "services";

let wrapper;
let mockProps = {
  threadCounts: { active: 17, idle: 8, stopped: 25, all: 40 },
  threadsFilter: "active"
};
const ThreadsTableToolbarWithProps = <ThreadsTableToolbar {...mockProps} />;

describe("ThreadsTableToolbar component", () => {
  beforeEach(function() {
    wrapper = mount(ThreadsTableToolbarWithProps);
  });

  test("Matches the snapshot", () => {
    const ThreadsTableToolbarComponent = renderer
      .create(<ThreadsTableToolbar {...mockProps} />)
      .toJSON();
    expect(ThreadsTableToolbarComponent).toMatchSnapshot();
  });

  test("Renders correct components", () => {
    expect(wrapper.find(Toolbar).length).toBe(1);
    expect(wrapper.find(ToolbarCenter).length).toBe(1);
    expect(wrapper.find(ButtonGroup).length).toBe(1);
    expect(wrapper.find(Button).length).toBe(4);
  });
  test("Buttons displays correct number of threadCounts per status", () => {
    const buttons = wrapper.find(Button);
    expect(buttons.at(0).props().suffix).toBe(40);
    expect(buttons.at(1).props().suffix).toBe(17);
    expect(buttons.at(2).props().suffix).toBe(8);
    expect(buttons.at(3).props().suffix).toBe(25);
  });
});
