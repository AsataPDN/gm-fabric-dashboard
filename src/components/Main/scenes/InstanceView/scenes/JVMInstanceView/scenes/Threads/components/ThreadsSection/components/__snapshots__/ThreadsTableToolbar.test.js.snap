// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadsTableToolbar component Matches the snapshot 1`] = `
<div
  className="sc-iwsKbI ciYlsK"
>
  <div
    className="sc-gZMcBi kaHcTN"
  >
    <div
      className="sc-dnqmqq ksISNG"
    >
      <button
        className="btn"
        disabled={false}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        title="All Threads"
      >
        <span
          className="label"
        >
          
          All Threads
          <span
            className="label-suffix"
          >
            40
          </span>
        </span>
      </button>
      <button
        className="btn active"
        disabled={false}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        title="Active"
      >
        <span
          className="label"
        >
          
          Active
          <span
            className="label-suffix"
          >
            17
          </span>
        </span>
      </button>
      <button
        className="btn"
        disabled={false}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        title="Idle"
      >
        <span
          className="label"
        >
          
          Idle
          <span
            className="label-suffix"
          >
            8
          </span>
        </span>
      </button>
      <button
        className="btn"
        disabled={false}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        title="Stopped"
      >
        <span
          className="label"
        >
          
          Stopped
          <span
            className="label-suffix"
          >
            25
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
